src/components/Layout/Navbar/components/UserProfile.tsx

```diff
@@ -1,34 +1,43 @@
-import { useGetSelfDetailsQuery } from "../../../../services/serverApi";
 import { Box, Text } from "@chakra-ui/react";
 import Image from "next/image";
-
+import { useSelector } from "react-redux";
+import { RootState } from "@/src/store";
 import { Dispatch, SetStateAction } from "react";
 
+import dummyImage from "../../../../../public/images/Avatar.png";
+
 import styles from "./userProfile.module.css";
 
-export default function UserProfile({ isDropdownVisible, setIsDropdownVisible} : {
+export default function UserProfile({
+  isDropdownVisible,
+  setIsDropdownVisible,
+}: {
   isDropdownVisible: boolean;
   setIsDropdownVisible: Dispatch<SetStateAction<boolean>>;
 }) {
-  const { data: user, isLoading } = useGetSelfDetailsQuery();
-  const imageToShow = user?.picture?.url || '/images/Avatar.png';
+  const { first_name, imageURL } = useSelector(
+    (state: RootState) => state.global
+  );
```

Comment:
- instead of selecting the entire global store which will re-renders even if `first_name` & `imageUrl` do not change can be more specific and pick first name and image url using the selector? 

```js
example:
---
useSelector(state => ({ firstName: state.global.firstName, imageUrl: state.global.imageUrl })
```

Reply:
This component is being deprecated in this PR  https://github.com/Real-Dev-Squad/members-site/pull/121